### PR TITLE
Change PythonInterperter VirtualEnv Behavior

### LINKED ISSUE #450: No error when Python dependencies are not found
Python kernel should probably fail when python dep isn't found (matching scala kernel behavior)

### LINKED ISSUE #921: pip dependency caching
Currently Polynote's pip dependency setup is very naive - it will just pip install whichever dependency you put. This leads to a couple problems (mostly if you are using wheel files rather than a full dependency): 

1. dependencies are cached and there's no way to bust the cache if you need to
2. old dependencies are not cleaned up (e.g., if you added `foo-v1` and now want to use `foo-v2`

### PR DESCRIPTION
**Pull request by jonathanindig**
* Clear venv when dependencies change (resolves #921)
* Error when dependencies are not found (resolves #450)
Screen Recording 2021-02-23 at 2 11 11 PM mov

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
-import zio.internal.{ExecutionMetrics, Executor}
+import zio.internal.Executor
 import zio.blocking.{Blocking, effectBlocking}
 import zio.{Has, RIO, Runtime, Task, UIO, ZIO}
 import zio.interop.catz._
 
 import scala.collection.JavaConverters._
-import scala.collection.mutable.ListBuffer
 import scala.concurrent.ExecutionContext
+import scala.reflect.io.Directory
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala:33]: TODO: replace usage of `import scala.reflect.io.Directory`

Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala:36]: Done

File: polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
+import zio.internal.Executor
 import zio.blocking.{Blocking, effectBlocking}
+import zio.duration._
 import zio.{Has, RIO, Runtime, Task, UIO, ZIO}
 import zio.interop.catz._
 
 import scala.collection.JavaConverters._
-import scala.collection.mutable.ListBuffer
 import scala.concurrent.ExecutionContext
+import scala.reflect.io.Directory
```
Comment (jeremyrsmith) [inline polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala:36]: Can you remove this import if it's no longer used?

Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala]: Done!

**Review by jeremyrsmith — APPROVED** [review]

**Pull request merged**